src/acceptance-test/java/org/zalando/nakadi/repository/db/EventTypeCacheTestAT.java

```diff
@@ -129,13 +145,106 @@ public void invalidateCacheOnUpdate() throws Exception {
+        final EventTypeCache etc = new RepositoriesConfig()
+                .eventTypeCache(client, eventTypeRepository, timelineRepository, timelineSync);
+        final EventType et = buildDefaultEventType();
+
+        Mockito.when(timelineRepository.listTimelines(et.getName())).thenReturn(getMockedTimelines(et.getName()));
+        Mockito.doReturn(et).when(eventTypeRepository).findByName(et.getName());
+
+        etc.created(et.getName());
+        final Optional<Timeline> timeline = etc.getActiveTimeline(et.getName());
+        Assert.assertEquals(Integer.valueOf(1), timeline.get().getOrder());
+    }
+
+    @Test
+    public void testGetTimelines() throws Exception {
+        final EventTypeCache etc = new RepositoriesConfig()
+                .eventTypeCache(client, eventTypeRepository, timelineRepository, timelineSync);
+        final EventType et = buildDefaultEventType();
+
+        Mockito.when(timelineRepository.listTimelines(et.getName())).thenReturn(getMockedTimelines(et.getName()));
+        Mockito.doReturn(et).when(eventTypeRepository).findByName(et.getName());
+
+        final List<Timeline> timelines = etc.getTimelines(et.getName());
+        Assert.assertEquals(3, timelines.size());
+    }
+
+    @Test
+    public void invalidateCacheOnTimelineChange() throws Exception {
+        final TestTimelineSync timelineSync = new TestTimelineSync();
+        final EventTypeCache etc = new RepositoriesConfig()
+                .eventTypeCache(client, eventTypeRepository, timelineRepository, timelineSync);
+        final EventType et = buildDefaultEventType();
+
+        Mockito.when(timelineRepository.listTimelines(et.getName())).thenReturn(getMockedTimelines(et.getName()));
+        Mockito.doReturn(et).when(eventTypeRepository).findByName(et.getName());
+
+        etc.created(et.getName());
+        timelineSync.invokeListeners();
+
+        executeWithRetry(() -> {
+                    try {
+                        etc.getTimelines(et.getName());
+                        verify(timelineRepository, times(2)).listTimelines(et.getName());
                     } catch (final Exception e) {
                         fail();
                     }
                 },
                 new RetryForSpecifiedTimeStrategy<Void>(5000).withExceptionsThatForceRetry(AssertionError.class)
                         .withWaitBetweenEachTry(500));
+    }
+
+    private List<Timeline> getMockedTimelines(final String etName) {
+        final Timeline t1 = new Timeline(etName, 0, new Storage(), "topic", new Date());
+        final Timeline t2 = new Timeline(etName, 1, new Storage(), "topic", new Date(System.currentTimeMillis() + 200));
+        t2.setSwitchedAt(new Date(System.currentTimeMillis() + 300));
+        final Timeline t3 = new Timeline(etName, 2, new Storage(), "topic", new Date(System.currentTimeMillis() + 500));
+        return ImmutableList.of(t1, t2, t3);
+    }
+
+    private class TestTimelineSync implements TimelineSync {
```

Comment:
This stub is used only for specific case when I want to call subscribes in order to invalidate cache.